src/Cookie.php

```diff
@@ -4,148 +4,150 @@
 
-   public function __construct(string $name)
-   {
-        $this->name = $name;
-        $this->value = '';
-        $this->expires = 0;
-        $this->path = '/';
-        $this->domain = null;
-        $this->secure = false;
-        $this->httpOnly = true;
-        $this->options = [];
-    }
-
-    public static function fromString(string $name): self {
-        return new self($name);
-    }
+	public function __construct(
+		string $name,
+		string $value = '',
+		int $expires = 0,
+		string $path = '/',
+		?string $domain = null,
+		bool $secure = false,
+		bool $httpOnly = false,
+		array $options = []
+	) {
+		$this->name = $name;
+		$this->value = $value;
+		$this->expires = $expires;
+		$this->path = $path;
+		$this->domain = $domain;
+		$this->secure = $secure;
+		$this->httpOnly = $httpOnly;
+		$this->options = $options;
+	}
 
-    public function getName(): string {
-        return $this->name;
-    }
+	public function getName(): string {
+		return $this->name;
+	}
 
-    public function setName(string $name): void {
-        if ('' === $name) {
-            return;
-        }
+	public function setName(string $name): void {
+		if ('' === $name) {
+			return;
+		}
 
 		$this->name = $name;
-    }
+	}
 
-    public function getValue(): string {
-        return $this->value;
-    }
+	public function getValue(): string {
+		return $this->value;
+	}
 
-    public function setValue(string $value): void {
-        $this->value = $value;
-    }
+	public function setValue(string $value): void {
+		$this->value = $value;
+	}
 
-    public function getExpires(): int {
-        return $this->expires;
-    }
+	public function getExpires(): int {
+		return $this->expires;
+	}
 
-    public function setExpires(int $expires): void {
-        $this->expires = $expires;
-    }
+	public function setExpires(int $expires): void {
+		$this->expires = $expires;
+	}
 
-    public function getPath(): string {
-        return $this->path;
-    }
+	public function isExpired(): bool {
+		return $this->expires < time();
+	}
 
-    public function setPath(string $path): void {
-        $this->path = $path;
-    }
+	public function getRemainingTime(): int {
+		return $this->expires - \time();
+	}
 
-    public function getDomain(): string {
-        return $this->domain;
-    }
+	public function getPath(): string {
+		return $this->path;
+	}
 
-    public function setDomain(string $domain): void {
-        $this->domain = $domain;
-    }
+	public function setPath(string $path): void {
+		$this->path = $path;
+	}
 
-    public function getSecure(): bool {
-        return $this->secure;
-    }
+	public function getDomain(): string {
+		return $this->domain;
+	}
 
-    public function setSecure(bool $secure): void {
-        $this->secure = $secure;
-    }
+	public function setDomain(string $domain): void {
+		$this->domain = $domain;
+	}
 
-    public function getHttpOnly(): bool {
-        return $this->httpOnly;
-    }
+	public function isSecure(): bool {
```

Comment:
`getSecure` should be `isSecure`. GET-Methods which returns bool normally starts with is instead of get.